src/Mollie/WC/Plugin.php

```diff
@@ -321,20 +326,65 @@ public static function maybeTestModeNotice()
 
     public static function maybeLetWCCancelOrder($willCancel, $order) {
         if (!empty($willCancel)) {
-            if ($order->get_payment_method()
-                !== 'mollie_wc_gateway_banktransfer'
+            $isMollieGateway = mollieWooCommerceIsMollieGateway($order->get_payment_method(
+            ));
+
+            $mollieDueDateEnabled = mollieWooCommerceIsGatewayEnabled($order->get_payment_method(
+            ), 'activate_expiry_days_setting');
+            if (!$isMollieGateway || !$mollieDueDateEnabled
             ) {
                 return $willCancel;
             }
-            //is banktransfer due date setting activated
-            $dueDateActive = mollieWooCommerceIsGatewayEnabled('mollie_wc_gateway_banktransfer_settings', 'activate_expiry_days_setting');
-            if ($dueDateActive) {
-                return false;
-            }
+
+            return false;
         }
         return $willCancel;
     }
 
+    public static function cancelOrderOnExpiryDate ()
+    {
+        $minHeldDuration = PHP_INT_MAX;
+        foreach (self::$GATEWAYS as $gateway){
+            $gatewayName = strtolower($gateway).'_settings';
+            $gatewaySettings = get_option( $gatewayName );
+            $heldDuration = isset($gatewaySettings['order_dueDate'])?$gatewaySettings['order_dueDate']:0;
+
+            if ( $heldDuration < 1 ) {
+                continue;
+            }
+            if($heldDuration < $minHeldDuration){
+                $minHeldDuration = $heldDuration;
+            }
+            $heldDurationInSeconds = $heldDuration*60;
+            if($gateway == 'Mollie_WC_Gateway_BankTransfer'){
+                $durationInHours = absint( $heldDuration )*24;
+                $durationInMinutes = $durationInHours*60;
+                $heldDurationInSeconds = $durationInMinutes*60;
+            }
+            $args = [
+                    'limit' => -1,
+                    'status' => 'pending',
+                    'payment_method' => strtolower($gateway),
+                    'date_modified' => '<' . (time() - $heldDurationInSeconds),
+            ];
+            $unpaid_orders =wc_get_orders( $args );
+
+            if ( $unpaid_orders ) {
+                foreach ( $unpaid_orders as $unpaid_order ) {
+                    $order = wc_get_order( $unpaid_order );
+                    add_filter('mollie-payments-for-woocommerce_order_status_cancelled', function($newOrderStatus){
+                        return Mollie_WC_Gateway_Abstract::STATUS_CANCELLED;
+                    });
+                    $order->update_status( 'cancelled', __( 'Unpaid order cancelled - time limit reached.', 'woocommerce' ), true );
+                    self::cancelOrderAtMollie( $order->get_id() );
+                }
+            }
+        }
+
+        wp_clear_scheduled_hook( 'woocommerce_cancel_unpaid_orders' );
+        wp_schedule_single_event( time() + ( absint( $minHeldDuration ) * 60 ), 'woocommerce_cancel_unpaid_orders' );
```

Comment:
Oh, I'm not sure about this part, but I guess it may cause problems. 

Let's imagine the case when `$minHeldDuration * 60` > WooCommerce setting for cancelling unpaid orders. Then, non-Mollie orders, which had to be removed on some date, will be removed later. So, we are breaking WC or other payment gateways.

If it was decided to not stick to WooCommerce settings, but create our own ones for cancelling orders, then I think it makes sense to create our own independent schedule (probably, daily).